modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesJMSPublisher.java

```diff
@@ -0,0 +1,353 @@
+    /**
+     * The logger used in logging information, warnings, errors and etc.
+     */
+    private static Logger log = Logger.getLogger(AndesJMSPublisher.class);
+
+    /**
+     * The configuration for the publisher
+     */
+    private AndesJMSPublisherClientConfiguration publisherConfig;
+
+    /**
+     * The amount of messages sent by the publisher
+     */
+    private long sentMessageCount;
+
+    /**
+     * The timestamp at which the first message was published
+     */
+    private long firstMessagePublishTimestamp;
+
+    /**
+     * The timestamp at which the last message was published
+     */
+    private long lastMessagePublishTimestamp;
+
+    /**
+     * The connection which is used to create the JMS session
+     */
+    private Connection connection;
+
+    /**
+     * The session which is used to create the JMS message producer
+     */
+    private Session session;
+
+    /**
+     * The message producer which produces/sends messages
+     */
+    private MessageProducer sender;
+
+    /**
+     * Message content which is needed to be published. The value will depend on the configuration.
+     */
+    private String messageContentFromFile = null;
+
+    /**
+     * Creates a new JMS publisher with a given configuration.
+     *
+     * @param config          The configuration
+     * @param createPublisher Creates connection, session and sender.
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public AndesJMSPublisher(AndesJMSPublisherClientConfiguration config, boolean createPublisher)
+            throws NamingException, JMSException {
+        super(config);
+
+        // Sets the configuration
+        this.publisherConfig = config;
+
+        // Creates a JMS connection, sessions and sender
+        if (createPublisher) {
+            ConnectionFactory connFactory = (ConnectionFactory) super.getInitialContext().lookup(AndesClientConstants.CF_NAME);
+            connection = connFactory.createConnection();
+            connection.start();
+            this.session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+            Destination destination = (Destination) super.getInitialContext().lookup(this.publisherConfig.getDestinationName());
+            this.sender = this.session.createProducer(destination);
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void startClient() throws JMSException, NamingException, IOException {
+        if (null != connection && null != session && null != sender) {
+            //reading message content from file
+            if (null != this.publisherConfig.getReadMessagesFromFilePath()) {
+                this.getMessageContentFromFile();
+            }
+
+            Thread subscriberThread = new Thread(this);
+            subscriberThread.start();
+        } else {
+            throw new NullPointerException("The connection, session and message sender is not assigned.");
```

Comment:
Any specific reason to throw an NPE here ? if yes, it should be added to the method signature